common/nymcoconut/src/proofs/mod.rs

```diff
@@ -0,0 +1,691 @@
+                .chain(std::iter::once(verification_key.alpha.to_bytes().as_ref()))
+                .chain(beta_bytes.iter().map(|b| b.as_ref()))
+                .chain(std::iter::once(commitment_kappa.to_bytes().as_ref()))
+                .chain(std::iter::once(commitment_zeta.to_bytes().as_ref())),
+        );
+
+        challenge == self.challenge
+    }
+
+    // challenge || rm.len() || rm || rt
+    pub(crate) fn to_bytes(&self) -> Vec<u8> {
+        //let attributes_len = self.response_attributes.len() as u64;
+        let attributes_len = 2;
+        let mut bytes = Vec::with_capacity((attributes_len + 1) as usize * 32);
+
+        bytes.extend_from_slice(&self.challenge.to_bytes());
+        bytes.extend_from_slice(&self.response_serial_number.to_bytes());
+        bytes.extend_from_slice(&self.response_binding_number.to_bytes());
+
+        // bytes.extend_from_slice(&attributes_len.to_le_bytes());
+        // for rm in &self.response_attributes {
+        //     bytes.extend_from_slice(&rm.to_bytes());
+        // }
+
+        bytes.extend_from_slice(&self.response_blinder.to_bytes());
+
+        bytes
+    }
+
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Result<Self> {
+        // at the very minimum there must be a single attribute being proven
+        if bytes.len() < 32 * 3 || (bytes.len()) % 32 != 0 {
+            return Err(CoconutError::DeserializationInvalidLength {
+                actual: bytes.len(),
+                modulus_target: bytes.len(),
+                modulus: 32,
+                object: "kappa and zeta".to_string(),
+                target: 32 * 3 + 8,
+            });
+        }
+
+        let challenge_bytes = bytes[..32].try_into().unwrap();
+        let challenge = try_deserialize_scalar(
+            &challenge_bytes,
+            CoconutError::Deserialization("Failed to deserialize challenge".to_string()),
+        )?;
+
+        // let rm_len = u64::from_le_bytes(bytes[32..40].try_into().unwrap());
+        if bytes[32..].len() != (2 + 1) as usize * 32 {
+            return Err(
+                CoconutError::Deserialization(
+                    format!("Tried to deserialize proof of kappa and zeta with insufficient number of bytes provided, expected {} got {}.", (2 + 1) as usize * 32, bytes[32..].len())
+                )
+            );
+        }
+
+        let serial_number_bytes = &bytes[32..64].try_into().unwrap();
+        let response_serial_number = try_deserialize_scalar(
+            serial_number_bytes,
+            CoconutError::Deserialization("failed to deserialize the serial number".to_string()),
+        )?;
+
+        let binding_number_bytes = &bytes[64..96].try_into().unwrap();
+        let response_binding_number = try_deserialize_scalar(
+            binding_number_bytes,
+            CoconutError::Deserialization("failed to deserialize the binding number".to_string()),
+        )?;
+        // let rm_end = 40 + rm_len as usize * 32;
```

Comment:
Since we don't seem to deserialize this anymore, please make sure the bounds on the byte slice are still valid, i.e. it's impossible to cause runtime panic by trying to read out of bounds